tests/CppUTestExt/MockSupportTest.cpp

```diff
@@ -256,6 +257,21 @@ TEST(MockSupportTestWithFixture, ShouldNotCrashOnFailureAfterCrashMethodWasReset
     CHECK_FALSE(cpputestHasCrashed);
 }
 
+TEST(MockSupportTestWithFixture, shouldCrashOnFailureWithCppUTestSetting)
+{
+    cpputestHasCrashed = false;
+    fixture.getRegistry()->setCrashOnFail();
+    UtestShell::setCrashMethod(crashMethod);
+    fixture.setTestFunction(unexpectedCallTestFunction_);
+
+    fixture.runAllTests();
+
+    CHECK(cpputestHasCrashed);
+
+    fixture.getRegistry()->setCrashOnFail(false);
+    UtestShell::resetCrashMethod();
+}
+
```

Comment:
Test that setting crash-on-fail in the registry does cause failing tests to crash